zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -505,65 +544,62 @@ impl<F: Field> SignVerifyChip<F> {
 
         let pk_le = pk_bytes_le(&sign_data.pk);
         let pk_be = pk_bytes_swap_endianness(&pk_le);
-        let pk_hash = (!padding)
+        let mut pk_hash = (!padding)
             .then(|| {
                 let mut keccak = Keccak::default();
                 keccak.update(&pk_be);
                 let hash: [_; 32] = keccak.digest().try_into().expect("vec to array of size 32");
                 hash
             })
-            .unwrap_or_default()
-            .map(|byte| Value::known(F::from(byte as u64)));
-        let pk_hash_hi = pk_hash[..12].to_vec();
+            .unwrap_or_default();
+
         // Ref. spec SignVerifyChip 2. Verify that the first 20 bytes of the
         // pub_key_hash equal the address
-        let (address, pk_hash_lo) = {
+        let (address, is_address_zero, is_address_zero_single) = {
```

Comment:
I think below function can be further simplify, right now a bit confusing.
Idea is split address into lo/hi, but keep `is_address_zero` just on single column, refer below proposed change
https://github.com/privacy-scaling-explorations/zkevm-circuits/compare/feat/%231387-TxCircuit-Refactor-word_rlc-into-word_lo/hi...feat/sm-test?expand=1

Reply:
thanks for your fix, I merge it directly